### ISSUE TITLE
[FR] Promote Appflowy Priovacy Better and tuchy features by showing more options (but useful) to the users.

### ISSUE THREAD

**Issue by salmanub** [AppFlowy-IO/AppFlowy#4529]
### Description

Just a proposal that remembers why Appflowy started its path.

### Impact

¿?

**Comment by annieappflowy** [comment]
Could you elaborate more? 
Specific features with use cases would be very helpful.

**Comment by diegovskytl** [comment]
I think it is related to the login options the app has right now.
Google, Github and Discord have bad privacy policies and using them to sync AppFlowy databases defeats part of the purpose of using the app.
Maybe clarifying more about how does the encrypted database behave inside Google, Github and Discord accounts can make it clearer to the users what is happening to their info.
I would suggest also considering the following points:
- in the AppFlowys privacy policy, indicate that third-party login options have their own privacy policies.
- include a direct link to Google, Github and Discord Privacy policies in Appflowys privacy policy (as some other platforms do, AliExpress and many online stores come to my mind).
I am willing to search for other options. codeberg.org could be an option to be researched and tested.

A git repo could be an alternative for some people. I used one to sync databases between my pc and my mac, but had to switch to Github SSO when I started using mobile. So enabling choosing the local folder for the database in mobile may be important to use alternative sync options.

**Comment by annieappflowy** [comment]
Hi @diegovskytl , thanks for your feedback. Here is some context:
The initial version of AppFlowy Mobile was released on TestFlight and Play Console just a few weeks ago. We implemented third-party logins first for two main reasons:

1. Quick implementation.
2. The assumption that most users would already have one of these accounts.

We are currently working on enabling direct login via password/magic link.
We'll also update the privacy policies. 
We're open to alternative login options.

**Comment by tio-trom** [comment]
Agree! I started to use AppFlowy because it is "privacy-oriented" but now there are too many non-privacy-friendly decisions. The integration of OpenAI chatbot, the integration of Google, Discord and the like.... I don't get it... There are FOSS alternatives for all of these. [GPT4ALL](https://github.com/nomic-ai/gpt4all) with its many open source language models. And Nextcloud for the cloud....

I am disappointed to be honest. I thought the focus is to make it more privacy friendly, more local, more control. 

I am afraid you will then create your own cloud and sell subscriptions for the sync, and that entanglement can never be good. 

I do appreciate your work a lot, but I had to say these.

**Comment by annieappflowy** [comment]
Hi @tio-trom,

We greatly appreciate your feedback, and I understand your preference for FOSS alternatives. In the spirit of transparency and honesty, I'm writing to provide more context about the features we've developed. Our mission is to enable everyone to achieve more with secure workplace tools. By default, OpenAI is turned off, allowing you to opt in with your own OpenAI key. This feature is designed for those who wish to enjoy the benefits of OpenAI without the need for tool switching. Unlike some alternatives to AppFlowy (such as Notion), AI in AppFlowy is turned off unless you decide to activate it. This decision was made with privacy in mind. We also integrate StabilityAI, powered by its open-source project, enabling users to generate images from text. This too is turned off by default. If you're not a big fan of AI, we won't force you to use it or to be the training data for AI. On the other hand, we want to offer AI to AppFlowy users who can benefit from its capabilities. If I remember correctly, Llama wasn't open-sourced when we integrated OpenAI. In the future, we plan to introduce more open-source alternative models into AppFlowy. However, please bear in mind that we can't do everything at once, given our small team size.

Regarding login options, as mentioned earlier in [this thread](https://github.com/AppFlowy-IO/AppFlowy/issues/4529#issuecomment-1913981500), we chose the most common third-party logins to get the initial version up and running. We're working on adding more login options. Having received constructive feedback from the community, we recognize in retrospect that our decision was not optimal, and we are working on improvements.

To be honest, I don't think it's fair to say that we're not prioritizing privacy, localization, and control. We offer a 100% offline experience, allowing users to locate, move, and back up their data folder as they see fit. We've enabled self-hosting for AppFlowy, giving you full control over your data in the cloud. We don't force users to use AI unless they opt in.

Regarding the cloud statement, if you're not a fan of AppFlowy Cloud, you have the option to use AppFlowy locally only and back up files to iCloud or OneDrive. Alternatively, you can self-host AppFlowy using your preferred cloud technology. The choice is yours.

In conclusion, I have a question for you. Apart from SaaS models, are there other revenue-generation strategies you would suggest for the AppFlowy project? We're working full-time and need to support ourselves and our families. We also hope to make the project sustainable so we can continue to improve AppFlowy for you.

**Comment by diegovskytl** [comment]
@tio-trom 
we must agree that popular services like github login and open ai can help AF's popularity. But let's help them constructively with their roadmap to ensure the projects sustainability while pushing more FOSS, privacy friendly characteristics.
> Apart from SaaS models, are there other revenue-generation strategies you would suggest for the AppFlowy project?
A suscription to have full-sync capabilities would be very nice, and a nicer alternative to recurrent donation (which could also be implemented). Also, having an obsidian-like one-time payment option that earns you some kind of badge would also be cool. Finally, maybe

**Comment by tio-trom** [comment]
@annieappflowy Thanks for the input. I would highly suggest for you to look at [GPT4ALL](https://github.com/nomic-ai/gpt4all) and add it as an option, at least, if possible. Would be great to not rely on these huge companies who can never be trusted, let's be honest. I know that the AI feature is disabled in AppFlowy, but it is still there, in a way promoted by a "privacy oriented" note taking app. That while there seem to be viable FOSS alternatives.

> To be honest, I don't think it's fair to say that we're not prioritizing privacy, localization, and control. We offer a 100% offline experience, allowing users to locate, move, and back up their data folder as they see fit. We've enabled self-hosting for AppFlowy, giving you full control over your data in the cloud. We don't force users to use AI unless they opt in.

True and for now AppFlowy is still doing ok in this regards. But I am truly concerned for the future of this project when dancing with such initiatives like inviting in, little by little, the big companies and financing models. I've seen several Note taking apps who rely on similar models, providing paid-for options to either unlock "pro" features or use their cloud, and then I can see their focus shifting towards not putting too much effort into their Open Source model for their cloud solution (self hosting) but more effort into their "pro" things, since they relied on them to keep the project alive.

And that's unfortunate. I had to remove one of such popular self hosted note taking app from our server, for this reason. They made it more and more difficult to self host and they publicly admitted it was because making it too easy would make people self host more and more and thus not use their cloud solution, so they would lose money.

It is a terrible situation that I've seen too many times. All because of a backwards society where people can't even focus on doing important and interesting projects, without worrying about how to survive.

What is the solution? Difficult to say, but you should at least try to rely on donations and see how it goes. There are some successful projects out there that do it like that.

**Comment by annieappflowy** [comment]
Supported login via email, self-hosting login on mobile,  and local mode
Working in progress: switch cloud AI models and local AI 
![image](https://github.com/AppFlowy-IO/AppFlowy/assets/12026239/ad3c129b-69a7-4486-bff3-aee49ef7de08)
https://github.com/AppFlowy-IO/AppFlowy/pull/5575

So I consider this issue was resolved. Please create a new one if you have other feedback or requests.
